Lib9c/Action/ClaimItems.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Bencodex.Types;
+using Libplanet.Action;
+using Libplanet.Action.State;
+using Libplanet.Crypto;
+using Libplanet.Types.Assets;
+using Nekoyume.Extensions;
+using Nekoyume.Model.Item;
+using Nekoyume.Model.State;
+using Nekoyume.TableData;
+using static Lib9c.SerializeKeys;
+
+namespace Nekoyume.Action
+{
+    public class ClaimItems : GameAction, IClaimItems
+    {
+        private const string ActionTypeText = "claim_item";
+
+        public IEnumerable<Address> AvatarAddresses { get; private set; }
+        public IEnumerable<FungibleAssetValue> Amounts { get; private set; }
+
+        public ClaimItems() {}
+
+        public ClaimItems(IEnumerable<Address> avatarAddresses, IEnumerable<FungibleAssetValue> amounts)
+        {
+            AvatarAddresses = avatarAddresses;
+            Amounts = amounts;
+        }
+
+        protected override IImmutableDictionary<string, IValue> PlainValueInternal =>
+            ImmutableDictionary<string, IValue>.Empty
+                .Add(AvatarAddressKey, new List(AvatarAddresses.Select(x => x.Serialize())))
+                .Add(AmountKey, new List(Amounts.Select(x => x.Serialize())));
+
+        protected override void LoadPlainValueInternal(
+            IImmutableDictionary<string, IValue> plainValue)
+        {
+            AvatarAddresses = ((List)plainValue[AvatarAddressKey]).Select(x => x.ToAddress());
+            Amounts = ((List)plainValue[AmountKey]).Select(x => x.ToFungibleAssetValue());
+        }
+
+        public override IAccount Execute(IActionContext context)
+        {
+            context.UseGas(1);
+
+            var states = context.PreviousState;
+            var itemSheet = states.GetSheets(containItemSheet: true).GetItemSheet();
+
+            var inventories = AvatarAddresses.Select(address =>
+            {
+                var addressHex = GetSignerAndOtherAddressesHex(context, address);
+                var inventory = states.GetInventory(address.Derive(LegacyInventoryKey));
+                if (inventory is null)
+                {
+                    throw new FailedLoadStateException(ActionTypeText, addressHex, typeof(AvatarState), address);
```

Comment:
Since the third argument is state's type, I think it should be `Inventory` instead of `AvatarState`.